tests/http_api_test.go

```diff
@@ -0,0 +1,148 @@
+package tests
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+type SomeJSONStruct struct {
+	SomeField string `json:"someField"`
+}
+
+func (s *clientIntegrationSuite) TestHTTPAPIBasics() {
+	// Create basic workflow that can answer queries, get signals, etc
+	workflowFn := func(ctx workflow.Context, arg *SomeJSONStruct) (*SomeJSONStruct, error) {
+		// Query that just returns query arg
+		err := workflow.SetQueryHandler(ctx, "some-query", func(queryArg *SomeJSONStruct) (*SomeJSONStruct, error) {
+			return queryArg, nil
+		})
+		if err != nil {
+			return nil, err
+		}
+		// Wait for signal to complete
+		var done bool
+		sel := workflow.NewSelector(ctx)
+		sel.AddReceive(workflow.GetSignalChannel(ctx, "some-signal"), func(ch workflow.ReceiveChannel, _ bool) {
+			var signalArg SomeJSONStruct
+			ch.Receive(ctx, &signalArg)
+			if signalArg.SomeField != "signal-arg" {
+				panic("invalid signal arg")
+			}
+			done = true
+		})
+		for !done {
+			sel.Select(ctx)
+		}
+		return arg, nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflowFn, workflow.RegisterOptions{Name: "http-basic-workflow"})
+
+	// Start
+	workflowID := s.randomizeStr("wf")
+	_, respBody := s.httpPost(http.StatusOK, "/api/v1/namespaces/"+s.namespace+"/workflows/"+workflowID, `{
+		"workflowType": { "name": "http-basic-workflow" },
+		"taskQueue": { "name": "`+s.taskQueue+`" },
+		"input": [{ "someField": "workflow-arg" }],
+		"requestId": "`+s.randomizeStr("req")+`"
+	}`)
+	var startResp struct {
+		RunID string `json:"runId"`
+	}
+	s.Require().NoError(json.Unmarshal(respBody, &startResp))
+
+	// Confirm already exists error with details and proper code
+	_, respBody = s.httpPost(http.StatusConflict, "/api/v1/namespaces/"+s.namespace+"/workflows/"+workflowID, `{
+		"workflowType": { "name": "http-basic-workflow" },
+		"taskQueue": { "name": "`+s.taskQueue+`" },
+		"input": [{ "someField": "workflow-arg" }],
+		"requestId": "`+s.randomizeStr("req")+`"
+	}`)
+	var errResp struct {
+		Message string `json:"message"`
+		Details []struct {
+			RunID string `json:"runId"`
+		} `json:"details"`
```

Comment:
For this style of details to work, I had to implement a custom error handler because the default reusing our marshaler doesn't work with Google "Any" wkt, and we want to use our marshaler.